stackrox/stackrox
# Delete obsolete attributes reducers and sagas

## Description

Warm up exercise after after a week away from the keyboard ;)

* reducers: 20 = 21 - 1 (counting network and policies folders as 1 reducer)
* sagas: = 10 = 11 - 1 

### Analysis of reducer

1. action creator `fetchUsersAttributes` function
    * sagas/attributesSagas calls `success` and `failure`

2. selector `getUsersAttributes` function has no calls

### Analysis of saga

For `accessControlPath` which is obsolete /main/access

### Analysis of accessControlPath

During cleanup after change from /main/access to /main/access-control I ran out of mental strength to confirm whether or not path matching is exact or partial in sagas:

* attributesSagas
    `takeEveryNewlyMatchedLocation(accessControlPath, getUsersAttributes)`
* groupSagas
    `takeEveryNewlyMatchedLocation(accessControlPath, getRuleGroups)`
* roleSagas
    `takeEveryNewlyMatchedLocation(accessControlPath, getRoles)`

### Residue

1. Remove old route path and rename new route paths.

2. Investigate console error at /main/access-control/auth-providers

    ```text
    console.js:35 Warning: Cannot update a component (`Connect(UserMenu)`) while rendering a different component (`AuthProviders`). To locate the bad setState() call inside `AuthProviders`, follow the stack trace as described in https://reactjs.org/link/setstate-in-render
    at AuthProviders
    at Route
    at Switch
    at Route
    at Switch
    at AccessControl
    at AsyncComponent
    at Route
    at Switch
    at ErrorBoundary
    ```

## Checklist

## Testing Performed

1. `yarn build` in ui

2. `wc build/static/js/*.js` in ui

    * branch - master: -1365 = 11682983 - 11684348 because webpack does not know that code is obsolete

3. `yarn start` in ui

4. Visit /main/access-control with `console.log` to verify no calls:

    * `usersAttributes` in reducers/attributes.js
    * `getUsersAttributes` in sagas/attributesSagas.js

5. Temporarily replace old with new access control path to see calls and verify that the absence of calls was not a false negative.

6. Visit /main/access-control with `console.log` to verify calls both before and after deleting obsolete access control path:
    * `getRuleGroups` in sagas/groupSagas.js
    * `getRoles` in sagas/roleSagas.js
